wallet/concepts/wallet-api.md

```diff
@@ -61,16 +61,19 @@ Make sure to handle errors for every call to
 ### Restricted methods
 
 MetaMask introduced wallet permissions in [EIP-2255](https://eips.ethereum.org/EIPS/eip-2255).
-In this permissions system, each RPC method is restricted or unrestricted.
-If a method is restricted, a dapp must request permission to call it using
-[`wallet_requestPermissions`](/wallet/reference/wallet_requestpermissions).
+In this permissions system, each RPC method is restricted or unrestricted. 
+
+Restricted methods are methods that cannot be called unless you have permission to do so using [`wallet_requestPermissions`](/wallet/reference/wallet_requestpermissions) or [wallet_requestSnaps](/wallet/reference/wallet_requestSnaps).
+
 Under the hood, permissions are plain, JSON-compatible objects, with fields that are mostly used
 internally by MetaMask.
 
-Outside of [Snaps restricted methods](/snaps/reference/rpc-api/#restricted-methods), the only
-restricted method is [`eth_accounts`](/wallet/reference/eth_accounts), which allows you to access
-the user's Ethereum accounts.
-More restricted methods will be added in the future.
+Outside of [Snaps restricted methods](/snaps/reference/rpc-api/#restricted-methods), there is the
+restricted method [`eth_accounts`](/wallet/reference/eth_accounts), which allows you to access
+the user's Ethereum accounts. DApps must explicitly request user consent using `wallet_requestPermissions` before using `eth_accounts`.
```

Comment:
Dapp

Reply:
As well as add a NOTE about eth_requestAccounts

Reply:
@alexandratran suggestion 

The following methods are restricted:
- eth_accounts - Requires passing eth_accounts as a parameter to wallet_requestPermissions. This unlocks:
  - eth_sendTransaction
  - personal_sign
  - eth_signTypedData_v4
note: we recommend calling eth_requestAccounts, which internally calls wallet_requestPermissions on eth_accounts
- wallet_snap - Requires calling wallet_requestSnap
- wallet_invokeSnap - Requires calling wallet_requestSnap


To call wallet_snap or wallet_invokeSnap on a Snap, you need to first call wallet_requestSnap on that Snap
Example on the bottom of this section: https://docs.metamask.io/snaps/learn/about-snaps/apis/#custom-json-rpc-apis (edited)